src/librustc_mir/transform/simplify.rs

```diff
@@ -371,16 +371,8 @@ impl<'a, 'tcx> Visitor<'tcx> for DeclMarker<'a, 'tcx> {
                 if let StatementKind::Assign(box (dest, rvalue)) = &stmt.kind {
                     if !dest.is_indirect() && dest.local == *local {
                         if let Rvalue::Use(Operand::Constant(c)) = rvalue {
-                            match c.literal.val {
-                                // Keep assignments from unevaluated constants around, since the
-                                // evaluation may report errors, even if the use of the constant
-                                // is dead code.
-                                ty::ConstKind::Unevaluated(..) => {}
-                                _ => {
-                                    trace!("skipping store of const value {:?} to {:?}", c, dest);
-                                    return;
-                                }
-                            }
+                            trace!("skipping store of const value {:?} to {:?}", c, dest);
```

Comment:
cc @wesleywiser Do you remember if we have any other hacks like this?

Reply:
This is the only one I know of.

Reply:
I've applied only this to the way the code looks now ...